src/main/java/io/pakland/mdas/githubstats/infrastructure/postgres/repository/PostgresMetricRepository.java

```diff
@@ -0,0 +1,48 @@
+package io.pakland.mdas.githubstats.infrastructure.postgres.repository;
+
+import io.pakland.mdas.githubstats.domain.Metric;
+import io.pakland.mdas.githubstats.domain.repository.MetricRepository;
+import io.pakland.mdas.githubstats.infrastructure.shell.model.ShellRequest;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.List;
+
+@Repository
+public interface PostgresMetricRepository extends JpaRepository<Metric, Integer>, MetricRepository {
```

Comment:
From our point of view (@manerow's and mine), this is not the correct approach as it is breaking Hex. Arch. rules.  
The thing is that we are explicitly telling our `app` layer, to use something, that is defined in the `infra` layer. However, this should not be possible, as communitcation goes the outter layers to inner layers.  

Instead, we should have a metric repository or something similar, that is definied inside the `domain.repository` package. Then, inside our `infra` layer, we should define the repository that will be implemented by JPA. We are not really familiar with JPA and we are not sure how to do it, but this is definitely not the correct way.

If this is not possible, we should somehow explain that in our report.